great_expectations/expectations/expectation.py

```diff
@@ -717,8 +725,27 @@ def _diagnostic_unexpected_statement_renderer(
         assert result, "Must provide a result object."
         success: Optional[bool] = result.success
         result_dict: dict = result.result
-
-        if result.exception_info["raised_exception"]:
+        exception: dict = {
+            "raised_exception": False,
+            "exception_message": "",
+            "exception_traceback": "",
+        }
+        if "raised_exception" in result.exception_info:
+            exception["raised_exception"] = result.exception_info["raised_exception"]
+            exception["exception_message"] = result.exception_info["exception_message"]
+            exception["exception_traceback"] = result.exception_info[
+                "exception_traceback"
+            ]
+        else:
+            for k, v in result.exception_info.items():
+                # TODO JT: This accounts for a dictionary of type {"metric_id": ExceptionInfo} path defined in
+                #  validator._resolve_suite_level_graph_and_process_metric_evaluation_errors
+                exception["raised_exception"] = v["raised_exception"]
+                exception["exception_message"] = v["exception_message"]
+                exception["exception_traceback"] = v["exception_traceback"]
+                break
```

Comment:
I think this `break` is where you are tossing out the retry exceptions? Does this give us the most recent one or the first one? Not sure that it matters. Can you add a comment here about why we need this `break`?

Reply:
Not the retry exceptions. I'm only including the first exception message which I not is not the most ideal thing but otherwise I would have had to come up with a way of rendering a list of exceptions in the data docs. I'll add a note about this.